data_model/src/query.rs

```diff
@@ -155,39 +144,47 @@ impl Payload {
     }
 }
 
-/// I/O ready structure to send queries.
-#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
-#[cfg(feature = "warp")]
-pub struct QueryRequest {
-    /// Payload
-    pub payload: Payload,
+model! {
+    /// I/O ready structure to send queries.
+    #[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
+    #[cfg(feature = "warp")]
+    #[serde(transparent)]
+    #[repr(transparent)]
+    pub struct QueryRequest {
+        /// Payload
+        pub payload: Payload,
+    }
 }
 
 #[cfg(feature = "warp")]
 declare_versioned_with_scale!(VersionedSignedQueryRequest 1..2, Debug, Clone, iroha_macro::FromVariant, IntoSchema);
 
-/// I/O ready structure to send queries.
-#[version_with_scale(n = 1, versioned = "VersionedSignedQueryRequest")]
-#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize, IntoSchema)]
-pub struct SignedQueryRequest {
-    /// Payload
-    pub payload: Payload,
-    /// Signature of the client who sends this query.
-    pub signature: SignatureOf<Payload>,
+model! {
+    /// I/O ready structure to send queries.
+    #[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[version_with_scale(n = 1, versioned = "VersionedSignedQueryRequest")]
+    pub struct SignedQueryRequest {
+        /// Payload
+        pub payload: Payload,
+        /// Signature of the client who sends this query.
+        pub signature: SignatureOf<Payload>,
+    }
 }
 
 declare_versioned_with_scale!(VersionedQueryResult 1..2, Debug, Clone, iroha_macro::FromVariant, IntoSchema);
 
 /// Sized container for all possible Query results.
-#[version_with_scale(n = 1, versioned = "VersionedQueryResult")]
 #[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[version_with_scale(n = 1, versioned = "VersionedQueryResult")]
+#[serde(transparent)]
+#[repr(transparent)]
 pub struct QueryResult(pub Value);
 
 declare_versioned_with_scale!(VersionedPaginatedQueryResult 1..2, Debug, Clone, iroha_macro::FromVariant, IntoSchema);
 
 /// Paginated Query Result
-#[version_with_scale(n = 1, versioned = "VersionedPaginatedQueryResult")]
 #[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[version_with_scale(n = 1, versioned = "VersionedPaginatedQueryResult")]
 pub struct PaginatedQueryResult {
```

Comment:
not wrapped into `model` atm, because client should have access to the fields of query result. Could be contended